graphql/rules.lua

```diff
@@ -204,7 +204,8 @@ function rules.argumentsOfCorrectType(node, context)
     for _, argument in pairs(node.arguments) do
       local name = argument.name.value
       local argumentType = parentField.arguments[name]
-      util.coerceValue(argument.value, argumentType.kind or argumentType)
+      util.coerceValue(argument.value, argumentType.kind or argumentType, nil, {
+          strict_non_null = true, skip_variables = true})
```

Comment:
nit: this table is created for each iteration but nobody modifies it. Let's create it only once.